### PR TITLE
feat: no recursion copy

### PR DESCRIPTION
**Pull request by tobymao**

### CODE REVIEW

File: sqlglot/expressions.py [hunk 1]
```diff
@@ -248,17 +248,37 @@ def meta(self) -> t.Dict[str, t.Any]:
...
+
+        while stack:
+            node, copy = stack.pop()
+
+            if node.comments is not None:
+                copy.comments = deepcopy(node.comments)
+            if node._type is not None:
+                copy._type = node._type.copy()
+            if node._meta is not None:
+                copy._meta = deepcopy(node._meta)
```
Comment (georgesittas) [inline sqlglot/expressions.py:263]: should we copy `self._hash` too?

**Review by georgesittas — APPROVED** [review]
Looks good, nice

File: sqlglot/expressions.py [hunk 2]
```diff
@@ -248,17 +248,37 @@ def meta(self) -> t.Dict[str, t.Any]:
...
+
+            for k, v in node.args.items():
+                if isinstance(v, Expression):
+                    c = v.__class__()
+                    copy.set(k, c)
+                    stack.append((v, c))
+                elif isinstance(v, list):
+                    for e in v:
```
Comment (georgesittas) [inline sqlglot/expressions.py:271]: FYI the failing test has to do with this branch - the edge case is `[]`. We need to copy that too.
